server/api/middleware.go

```diff
@@ -56,6 +68,99 @@ type entry struct {
 	value string
 }
 
+type componentMiddleware struct {
+	s  *server.Server
+	rd *render.Render
+}
+
+func newComponentMiddleware(s *server.Server) componentMiddleware {
+	return componentMiddleware{
+		s:  s,
+		rd: render.New(render.Options{IndentJSON: true}),
+	}
+}
+
+func (m componentMiddleware) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I wonder if it is suitable to use middleware here. After all, the logic here is too complicated.